utils/evaluate_utils.py

```diff
@@ -13,12 +13,83 @@
 
 from train_utils import get_X_y_data
 
+from sklearn.metrics import precision_recall_curve
+from sklearn.preprocessing import label_binarize
+
 # set numpy seed to make random operations reproduceable
 np.random.seed(0)
 
-def evaluate_model_cm(
-    log_reg_model: LogisticRegression, dataset: pd.DataFrame
-):
+
+def class_PR_curves(single_cell_data: pd.DataFrame, log_reg_model):
+    """
+    save precision recall curves for each class to the save directory
+    also, return the precision/recall data for each class in tidy long format
+
+    Parameters
+    ----------
+    single_cell_data : pd.DataFrame
+        single cell dataframe with correct cell metadata and feature data
+    log_reg_model : sklearn classifier
+        clasifier to get precision recall curves for
+
+    Returns
+    -------
+    pd.DataFrame
+        dataframe with precision/recall data for each class in tidy long format
+
+    matplotlib.figure.Figure
+        figure with compilation of all class PR curves
+    """
+
+    phenotypic_classes = log_reg_model.classes_
+    X, y = get_X_y_data(single_cell_data)
```

Comment:
If this takes a while to load, you can consider loading outside the function (in the notebook) and take X, y as input to the function. If it is quick, then no worries.

Reply:
This is quick :)